firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayPanelViewController.swift

```diff
@@ -182,7 +182,8 @@ class TabDisplayPanelViewController: UIViewController,
 
     private func retrieveTheme() -> Theme {
         if featureFlags.isFeatureEnabled(.feltPrivacySimplifiedUI, checking: .buildOnly) {
-            return themeManager.resolvedTheme(with: tabsState.isPrivateMode)
+            let isPrivateMode = tabsState.isPrivateMode && panelType == .privateTabs
```

Comment:
a bit worried about adding this check, since the state should be the single source of truth and glancing at how we set `isPrivate` it should be set by this check already `isPrivateMode: panelType == .privateTabs`. If we can't rely on the state, then there might be another underlying issue. What do you think?

Reply:
Yeah the issue is basically that we have multiple `tabState` event fired, and normal and private tabs panel both receive them from what I currently understand (not a redux expert yet). I asked Orla and she said they kind of faced that issue at the time when multi window was opened (so a redux event is firedand multiple objects can receive those events). That's why they introduced the `windowUUID`, so they can know which window should answer to which event. I was trying to work around that here, but seems like I need to find a better solution since there's some issues with this haha. Thanks for bringing that up!